testing-plugin/src/main/kotlin/arrow/meta/plugin/testing/CompilationAssertions.kt

```diff
@@ -16,35 +13,65 @@ private data class ExpressionParts(
   val property: String? = null
 )
 
-internal fun assertThis(compilationData: CompilationData): Result {
-  val compilationResult = compile(compilationData)
+val interpreter: (CompilerTest) -> Unit = {
+
+  fun runConfig(config: Config): Unit = when (config) {
+    is Config.AddCompilerPlugins -> println("Plugins found: ${config.plugins}")
+    is Config.AddDependencies -> println("Dependencies found: ${config.dependencies}")
+    is Config.Many -> config.configs.forEach(::runConfig)
+    Config.Empty -> println("Testing configuration not found")
+  }
 
-  assertThat(compilationResult.actualStatus).isEqualTo(compilationData.expectedStatus)
-  compilationData.checks.forEach {
-    when (it) {
-      is ExpectedCompilationError -> checkCompilationResult(compilationResult, it)
-      is ExpectedGeneratedSourceCode -> checkExpectedGeneratedSourceCode(compilationResult, it)
-    }
+  fun runAssert(assert: Assert, compilationResult: CompilationResult): Unit = when (assert) {
+    Assert.Empty -> println("Assertions not found")
+    Assert.CompilationResult.Compiles -> assertCompiles(compilationResult)
+    Assert.CompilationResult.Fails -> assertFails(compilationResult)
+    is Assert.CompilationResult.FailsWith -> assertFailsWith(compilationResult, assert.f)
+    is Assert.QuoteOutputMatches -> assertQuoteOutputMatches(compilationResult, assert.source)
+    is Assert.EvalsTo -> assertEvalsTo(compilationResult, assert.source, assert.output)
   }
-  return Result(classesDirectory = compilationResult.classesDirectory)
+
+  val configuration = it.config(CompilerTest)
+  val assertions = it.assert(CompilerTest)
+
+  configuration.map(::runConfig)
+
+  val compilerPlugins = configuration.filterIsInstance<Config.Many>().flatMap { it.configs }.filterIsInstance<Config.AddCompilerPlugins>().flatMap { it.plugins }.flatMap { it.dependencies }.map { it.mavenCoordinates }
```

Comment:
Let's pair on this part, this should be easier than this by folding or recursing over the cases.

Reply:
Thanks @raulraja ! :+1: